src/Arcus.Security.Providers.AzureKeyVault/Configuration/SecretStoreBuilderExtensions.cs

```diff
@@ -0,0 +1,100 @@
+            string clientId,
+            X509Certificate2 certificate)
+        {
+            Guard.NotNull(rawVaultUri, nameof(rawVaultUri));
+            Guard.NotNull(clientId, nameof(clientId));
+            Guard.NotNull(certificate, nameof(certificate));
+
+            return AddAzureKeyVault(
+                builder,
+                new CertificateBasedAuthentication(clientId, certificate),
+                new KeyVaultConfiguration(rawVaultUri));
+        }
+
+        /// <summary>
+        /// Adds a secret source to the secret store of the application that gets its secrets from Azure Key Vault.
+        /// </summary>
+        /// <param name="builder">The builder to create the secret store.</param>
+        /// <param name="rawVaultUri">The Uri of the Azure Key Vault you want to connect to.</param>
+        /// <param name="connectionString">The connection string to use to authenticate, if applicable.</param>
+        /// <param name="azureADInstance">The azure AD instance to use to authenticate, if applicable.</param>
+        public static SecretStoreBuilder AddAzureKeyVaultWithManagedServiceIdentity(
+            this SecretStoreBuilder builder,
+            string rawVaultUri,
+            string connectionString = null,
+            string azureADInstance = null)
+        {
+            Guard.NotNull(rawVaultUri, nameof(rawVaultUri));
+
+            return AddAzureKeyVault(
+                builder,
+                new ManagedServiceIdentityAuthentication(connectionString, azureADInstance),
+                new KeyVaultConfiguration(rawVaultUri));
+        }
+
+        /// <summary>
+        /// Adds a secret source to the secret store of the application that gets its secrets from Azure Key Vault.
+        /// </summary>
+        /// <param name="builder">The builder to create the secret store.</param>
+        /// <param name="rawVaultUri">The Uri of the Azure Key Vault you want to connect to.</param>
+        /// <param name="clientId">The ClientId of the service principal, used to connect to Azure Key Vault</param>
+        /// <param name="clientKey">The Secret ClientKey of the service principal, used to connect to Azure Key Vault</param>
+        public static SecretStoreBuilder AddAzureKeyVaultWithServicePrincipal(
+            this SecretStoreBuilder builder,
+            string rawVaultUri,
+            string clientId,
+            string clientKey)
+        {
+            Guard.NotNull(rawVaultUri, nameof(rawVaultUri));
+            Guard.NotNullOrEmpty(clientId, nameof(clientId));
+            Guard.NotNullOrEmpty(clientKey, nameof(clientKey));
+
+            return AddAzureKeyVault(
+                builder,
+                new ServicePrincipalAuthentication(clientId, clientKey),
+                new KeyVaultConfiguration(rawVaultUri));
+        }
+
+        /// <summary>
+        /// Adds a secret source to the secret store of the application that gets its secrets from Azure Key Vault.
```

Comment:
Was more suggesting to use something similar to this:
```suggestion
        /// Adds Azure Key Vault as a secret source.
```